services/editor/src/hoc/with-loading.js

```diff
@@ -1,14 +1,30 @@
 import React from 'react';
 import { compose, lifecycle, withState } from 'recompose';
 
-const withLoading = (loadingRenderer, loadingPromiseFactory) => (Comp) =>
+const withLoading = (loadingRenderer, loadingErrorRenderer, loadingPromiseFactory) => Comp =>
   compose(
     withState('isLoading', 'setIsLoading', true),
+    withState('loadingError', 'setLoadingError', null),
     lifecycle({
-      componentWillMount() {
-        loadingPromiseFactory(this.props).then(() => this.props.setIsLoading(false));
+      async componentWillMount() {
```

Comment:
it's not the best practice to make `async componentWillMount`. 
can you do it in `componentDidMount`?